pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -450,19 +450,23 @@ protected CompletableFuture<Void> internalCreateNonPartitionedTopicAsync(boolean
                     // update remote cluster
                     return namespaceResources().getPoliciesAsync(namespaceName)
                             .thenCompose(policies -> {
-                                if (!policies.isPresent()) {
+                                if (policies.isEmpty()) {
                                     return CompletableFuture.completedFuture(null);
                                 }
                                 // Combine namespace level policies and topic level policies.
                                 Set<String> replicationClusters = policies.get().replication_clusters;
                                 TopicPolicies topicPolicies =
                                         pulsarService.getTopicPoliciesService().getTopicPoliciesIfExists(topicName);
-                                if (topicPolicies != null) {
-                                    replicationClusters = new HashSet<>(topicPolicies.getReplicationClusters());
+                                if (topicPolicies != null && topicPolicies.getReplicationClusters() != null) {
+                                    replicationClusters = new HashSet<>();
                                 }
```

Comment:
This change seems unnecessary.

Reply:
The `topicPolicies.getReplicationClusters()` may be null and it will throw NPE when creating the HashSet.
It can be reproduced when running the `testPartitionedShadowTopicExpansion`.